services/directions.js

```diff
@@ -0,0 +1,151 @@
+ * @param {boolean} [config.voice_instructions=false]
+ * @param {string} [config.voice_units=imperial]
+ * @return {MapiRequest}
+ */
+Directions.getDirections = function(config) {
+  config.profile = config.profile || 'driving';
+
+  v.validate(
+    {
+      profile: v.string,
+      way_points: v.arrayOfObjects.required,
+      alternatives: v.boolean,
+      annotations: v.arrayOfStrings,
+      banner_instructions: v.boolean,
+      continue_straight: v.boolean,
+      exclude: v.string,
+      geometries: v.string,
+      language: v.string,
+      overview: v.string,
+      roundabout_exits: v.boolean,
+      steps: v.boolean,
+      voice_instructions: v.boolean,
+      voice_units: v.string
+    },
+    config
+  );
+
+  config.way_points.forEach(function(wayPoint) {
+    v.validate(
+      {
+        latitude: v.number.required,
+        longitude: v.number.required,
+        approach: v.string,
+        bearing: v.arrayOf(v.number),
+        radius: v.oneOf([v.number, v.string]),
+        waypoint_name: v.string
+      },
+      wayPoint
+    );
+  });
+
+  var wayPoints = {
+    coordinates: [],
+    approach: [],
+    bearing: [],
+    radius: [],
+    waypoint_name: []
+  };
+
+  config.way_points.forEach(function(wayPoint) {
+    wayPoints.coordinates.push(wayPoint.longitude + ',' + wayPoint.latitude);
+
+    // join props which come in pairs
+    ['bearing'].forEach(function(prop) {
+      if (wayPoint.hasOwnProperty(prop) && wayPoint[prop] != null) {
+        wayPoint[prop] = wayPoint[prop].join(',');
+      }
+    });
+
+    ['approach', 'bearing', 'radius', 'waypoint_name'].forEach(function(prop) {
+      if (wayPoint.hasOwnProperty(prop) && wayPoint[prop] != null) {
+        wayPoints[prop].push(wayPoint[prop]);
+      } else {
+        wayPoints[prop].push('');
+      }
+    });
+  });
+
+  ['approach', 'bearing', 'radius', 'waypoint_name'].forEach(function(prop) {
+    // avoid sending params which are all `;`
+    if (
+      wayPoints[prop].every(function(char) {
+        return char == '';
+      })
+    ) {
+      delete wayPoints[prop];
+    } else {
+      wayPoints[prop] = wayPoints[prop].join(';');
+    }
+  });
+
+  var query = {
+    alternatives: config.alternatives,
+    annotations: config.annotations,
+    banner_instructions: config.banner_instructions,
+    continue_straight: config.continue_straight,
+    exclude: config.exclude,
+    geometries: config.geometries,
+    language: config.language,
+    overview: config.overview,
+    roundabout_exits: config.roundabout_exits,
+    steps: config.steps,
+    voice_instructions: config.voice_instructions,
+    voice_units: config.voice_units,
+
+    approach: wayPoints.approach,
+    bearing: wayPoints.bearing,
+    radius: wayPoints.radius,
+    waypoint_name: wayPoints.waypoint_name
+  };
+
+  // remove null/undefined keys
+  Object.keys(query).forEach(function(key) {
+    if (query[key] == null) {
+      delete query[key];
+    }
+  });
```

Comment:
Let's make a shareable utility function for this, in case it comes up again. Maybe something like `pickDefined`?